lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -1033,6 +1021,14 @@ const char16* WasmBinaryReader::ReadInlineName(uint32& length, uint32& nameLengt
         AssertMsg(UNREACHED, "We calculated the length before decoding, what happened ?");
         ThrowDecodingError(_u("Error while decoding utf8 string"));
     }
+    for (size_t i = 0; i < decodedLength; ++i)
+    {
+        const char16 c = contents[i];
+        if (!utf8::IsValidWideChar(c) || c == UNICODE_UNKNOWN_CHAR_MARK)
+        {
+            ThrowDecodingError(_u("Invalid UTF-8 encoding"));
```

Comment:
It sounds like you're saying this possibly-redundant check wouldn't really hurt performance? If so, you can leave it in. Otherwise, I believe it is correct so say that if you are looking for the presence of invalid characters in the utf-8 input buffer by looking at the utf-16 output, then looking for `UNICODE_UNKNOWN_CHAR_MARK` is the only thing that is necessary.